sklearn/gaussian_process/_gpc.py

```diff
@@ -262,18 +262,30 @@ def obj_func(theta, eval_gradient=True):
 
         return self
 
-    def predict(self, X):
+    def predict(self, X, return_std_of_f=False):
```

Comment:
I think that `std_of_f` should be available only via `predict_proba`.
As `predict` does not provide a probabilistic estimation returning an the std of the nuiance function is out of the scope in a way.

Reply:
Hi Noa,

First of all, thanks for the review!

I added the flag here as well to match the API for `GPR`. In the `predict` method for Gaussian Process Regression, you can also find a `return_std`. If we remove it from this `predict`, wouldn't that make the APIs different?

I'm happy to remove it either way, but I'm curious about what you think regarding whether the APIs for `GPR` and `GPC` should match.

Reply:
Maybe @adrinjalali could share his opinion of whether keeping the interfaces as similar as possible.

Reply:
If it's worth anything, someone mentioned the consistency with `GPR` in a comment to the original issue: https://github.com/scikit-learn/scikit-learn/issues/22226#issuecomment-1343028703

Reply:
I agree that in this case having it only for `predict_proba` makes more sense.

Reply:
I'll include those changes then!